src/HDF5.jl

```diff
@@ -1993,7 +2013,19 @@ function hdf5_to_julia_eltype(objtype)
         T = HDF5Vlen{hdf5_to_julia_eltype(HDF5Datatype(super_id))}
     elseif class_id == H5T_COMPOUND
         N = Int(h5t_get_nmembers(objtype.id))
-        T = HDF5Compound{N}
+        # check if should be interpreted as complex
+        if COMPLEX_SUPPORT[] && N == 2
+          membernames = ntuple(N) do i
+            h5t_get_member_name(objtype.id, i-1)
+          end
+          membertypes = ntuple(N) do i
+            h5t_get_member_type(objtype.id, i-1) |> HDF5Datatype |> hdf5_to_julia_eltype
```

Comment:
```suggestion
            hdf5_to_julia_eltype(HDF5Datatype(h5t_get_member_type(objtype.id, i-1)))
```
The rest of the code base doesn't use `|>` syntax so let's just stick to the convention ?

Reply:
done